public/app/features/transformers/extractFields/fieldExtractor.test.ts

```diff
@@ -14,6 +14,37 @@ describe('Extract fields from text', () => {
     `);
   });
 
+  it('Test key-values with single/double quotes', async () => {
+    const extractor = fieldExtractors.get(FieldExtractorID.KeyValues);
+    const out = extractor.parse('a="1",   "b"=\'2\',c=3  x:y ;\r\nz="d and 4"');
+    expect(out).toMatchInlineSnapshot(`
+      Object {
+        "a": "1",
+        "b": "2",
+        "c": "3",
+        "x": "y",
+        "z": "d and 4",
+      }
+    `);
+  });
+
+  it('Test key-values with nested single/double quotes', async () => {
+    const extractor = fieldExtractors.get(FieldExtractorID.KeyValues);
+    const out = extractor.parse(
+      `a="1",   "b"=\'2\',c=3  x:y ;\r\nz="dbl_quotes="Double Quotes" sgl_quotes='Single Quotes'"`
+    );
+
+    expect(out).toMatchInlineSnapshot(`
+      Object {
+        "a": "1",
+        "b": "2",
+        "c": "3",
+        "x": "y",
+        "z": "dbl_quotes=\\"Double Quotes\\" sgl_quotes=\\"Single Quotes\\"",
```

Comment:
`\\"Single Quotes\\""`

I think I'm seeing here that the single quotes are lost and replaced with double quotes.
What do you think about preserving the single quotes?

Reply:
We should be preserving single quotes, I could expand the regexp code to do this but I think @ryantxu's code may be more appropriate as it handles this case.